lib/using_restricted_array.rb

```diff
@@ -1,61 +1,118 @@
 require_relative 'restricted_array.rb'
+require 'pry'
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear or O(n), where n is the number of items in the array or the length of the input array
+# Space complexity:
 def length(array)
-  raise NotImplementedError
+  #delete NotImplementedError and write your code there instead
+  i = 0
+
+  until array[i] == nil
+    i += 1
+  end
+  return i
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
+# Time complexity: quadratic, or O(n^2), where n is the number of items in the array or the length of the input array
```

Comment:
`O(n^2)` usually (but not always) means two loops. Given that there's only one loop below, I reckon your code is _better_ than `O(n^2)`. 🙂